### PR TITLE
PartTemplate work

### PR DESCRIPTION
**Pull request by RossStewart**
PR for PartTemplate work. Initial backend submitted more submissions and tests to follow.

### CODE REVIEW

File: src/Facade/ResourceBuilders/PartTemplateResourceBuilder.cs [hunk 1]
```diff
@@ -15,10 +15,13 @@ public PartTemplateResource Build(PartTemplate model)
...
+                           AllowVariants = model.AllowVariants,
+                           Variants = model.Variants,
                            AccountingCompany = model.AccountingCompany,
                            ProductCode = model.ProductCode,
                            StockControlled = model.StockControlled,
                            LinnProduced = model.LinnProduced,
+                           RmFg = model.RmFg,
                            BomType = model.BomType,
                            AssemblyTechnology = model.AssemblyTechnology,
                            AllowPartCreation = model.AllowPartCreation,
```
Comment (richardiphillips) [inline src/Facade/ResourceBuilders/PartTemplateResourceBuilder.cs:27]: remember you can set the "self" linkrel in GetLocation down below

File: src/Persistence.LinnApps/Repositories/PartTemplateRepository.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ public IQueryable<PartTemplate> FindAll()
```
Comment (richardiphillips) [inline src/Persistence.LinnApps/Repositories/PartTemplateRepository.cs:35]: remember to take off AsNoTracking up above if we're amending these things now

**Review by richardiphillips — APPROVED** [review]
Looks good. Needs a couple of integration tests added for put and post i think.

**Comment by RossStewart** [comment]
No worries, was going to add a few more tests on Monday morning so I’ll make these cha ges then too!

File: src/Service.Host/client/src/containers/parts/Parts.js [hunk 3]
```diff
@@ -6,13 +6,16 @@ import partsSelectors from '../../selectors/partsSelectors';
 import { getPrivileges } from '../../selectors/userSelectors';
 import partTemplatesActions from '../../actions/partTemplatesActions';
 import partTemplatesSelectors from '../../selectors/partTemplatesSelectors';
+import partTemplateActions from '../../actions/partTemplatesActions';
+import partTemplateSelectors from '../../selectors/partTemplatesSelectors';
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/containers/parts/Parts.js:11]: do we need these extra action creators and selectors in the parts container?

Comment (lewisrenfrew) [reply src/Service.Host/client/src/containers/parts/Parts.js:11]: I also just realised I never got you set up witht the linter which would have complained about this most likely. Give me a shout today and I can try and remember what we need to do to get it working

Comment (RossStewart) [reply src/Service.Host/client/src/containers/parts/Parts.js:11]: Can talk about it after stand up if you're about

**Pull request merged**